test/deployframework/framework.go

```diff
@@ -0,0 +1,335 @@
+
+	envVarArr, err := df.createResourceDirs(pathToTestCaseOutputDir)
+	if err != nil {
+		logger.Warnf("failed to create the resource output directories: %v", err)
+	}
+
+	cleanupCmd := exec.Command(df.CleanupScriptPath)
+	cleanupStdout, err := cleanupCmd.StdoutPipe()
+	if err != nil {
+		logger.Warnf("Failed to create pipe from command output to Stdout: %v", err)
+	}
+
+	scanner := bufio.NewScanner(cleanupStdout)
+	go func() {
+		for scanner.Scan() {
+			line := scanner.Text()
+			logger.Infof(line)
+		}
+		if err := scanner.Err(); err != nil {
+			logger.Errorf("error reading output from command: %v", err)
+		}
+		return
+	}()
+
+	cleanupCmd.Env = append(os.Environ(), envVarArr...)
+	err = cleanupCmd.Run()
+	if err != nil {
+		logger.Warnf("error running the cleanup script: %v", err)
+	}
+
+	return df.Deployer.Uninstall()
+}
+
+type podStat struct {
+	PodName string
+	Ready   int
+	Total   int
+}
+
+// WaitForMeteringPods periodically polls the list of pods in the @namespace
+// and ensures the metering pods created are considered ready. In order to exit
+// the polling loop, the number of pods listed must match the expected number
+// of @targetPods, and all pod containers listed must report a ready status.
+func (df *DeployFramework) WaitForMeteringPods(targetPods int, namespace string) (bool, error) {
+	var readyPods []string
+	var unreadyPods []podStat
+
+	start := time.Now()
+
+	df.Logger.Infof("Waiting for all metering pods to be ready")
+	err := wait.Poll(10*time.Second, 20*time.Minute, func() (done bool, err error) {
+		unreadyPods = nil
+		readyPods = nil
+
+		pods, err := df.Client.CoreV1().Pods(namespace).List(meta.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if len(pods.Items) == 0 {
+			return false, fmt.Errorf("the number of pods in the %s namespace should exceed zero", namespace)
+		}
+
+		for _, pod := range pods.Items {
+			podIsReady, readyContainers := df.checkPodStatus(pod)
+			if podIsReady {
+				readyPods = append(readyPods, pod.Name)
+				continue
+			}
+
+			unreadyPods = append(unreadyPods, podStat{
+				PodName: pod.Name,
+				Ready:   readyContainers,
+				Total:   len(pod.Status.ContainerStatuses),
+			})
+		}
+
+		df.logPollingSummary(targetPods, readyPods, unreadyPods)
+
+		return len(pods.Items) == targetPods && len(unreadyPods) == 0, nil
+	})
+	if err != nil {
+		return false, fmt.Errorf("the metering pods failed to report a ready status before the timeout period occurred: %v", err)
+	}
+
+	df.Logger.Infof("Installing metering took %v", time.Since(start))
+
+	return true, nil
+}
+
+// GetRouteBearerToken queries the @namespace for the reporting-operator service account and attempts
+// to find the secret that contains the reporting-operator token. If that secret exists, return the
+// string representation of the token (key) byte slice (value), or return an error.
+func (df *DeployFramework) GetRouteBearerToken(namespace string) (string, error) {
```

Comment:
Minor nit: This is referenced in line 159, so maybe throwing this function more towards the beginning of the script would keep it clearer

Reply:
This isn't really that good of advice since usually functions can live in separate files from where they're used, so where it is in a file shouldn't really matter.